src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -128,9 +135,40 @@ export default class SurvivalChart extends React.Component<ISurvivalChartProps,
         let configurableOpts: StyleOpts = _.merge({}, this.styleOptsDefaultProps, this.props.styleOpts);
         configurableOpts.padding.right = this.props.showLegend ? 300 : configurableOpts.padding.right;
         configurableOpts.legend.x = configurableOpts.width - configurableOpts.padding.right;
+        this.downSamplingDenominatorX = configurableOpts.width - configurableOpts.padding.left - configurableOpts.padding.right;
```

Comment:
computed getters should be pure functions and should not have sideffects (setting instance properties).   it looks like these denomintator props should be computed properties.